pkg/volume/gce_pd/attacher.go

```diff
@@ -165,6 +165,9 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
 	options := []string{}
 	if readOnly {
 		options = append(options, "ro")
+	} else {
+		// as per https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
```

Comment:
oh!  Yeah, we could, I'm just wary of too many "is ext4" conditionals
spread throughout the code makeing assumptions about what flags other parts
did or did not use.

On Thu, Jul 7, 2016 at 12:39 AM, Michael Rubin <notifications@github.com>
wrote:

> In pkg/volume/gce_pd/attacher.go
> <https://github.com/kubernetes/kubernetes/pull/28448#discussion_r69863021>
> :
>
> > @@ -165,6 +165,9 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
> >  	options := []string{}
> >  	if readOnly {
> >  		options = append(options, "ro")
> > +	} else {
> > +		// as per https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
>
> @thockin <https://github.com/thockin> @ciwang <https://github.com/ciwang>
> @saad-ali <https://github.com/saad-ali>
> Can we invoke file -s <http://linux.die.net/man/1/file> on the block
> device so that we can determine the filesystem type in a portable manner?
> Check out: http://www.thegeekstuff.com/2011/04/identify-file-system-type/
> This might be a good PR for ciwang's stack of intern items. Or does "file"
> not exist in our image?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/28448/files/8efefab9a3fdc4906ca53d4a23d22c3846d0610b#r69863021>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFVgVGT2iHT8jQF0IdjDo2mBquiaRFSLks5qTK0rgaJpZM4JECrD>
> .
>